packages/sandbox/src/setupSandboxEmbedRelay.ts

```diff
@@ -90,13 +99,43 @@ export function setupSandboxEmbedRelay({
             operationId,
             embedUrl,
           });
+        } else if (isSubscription) {
+          checkForProtocolChange = (event: MessageEvent) => {
+            if (event.data.name === EXPLORER_CHANGE_PROTOCOL) {
+              executeSubscription({
+                operation,
+                operationName,
+                variables,
+                headers,
+                embeddedIFrameElement: embeddedSandboxIFrameElement,
+                operationId,
+                embedUrl,
+                subscriptionUrl: data.subscriptionUrl,
+                protocol: event.data.protocol,
+              });
+            }
+          };
+          executeSubscription({
+            operation,
+            operationName,
+            variables,
+            headers,
+            embeddedIFrameElement: embeddedSandboxIFrameElement,
+            operationId,
+            embedUrl,
+            subscriptionUrl: data.subscriptionUrl,
+            protocol: data.protocol,
+          });
         }
       }
     }
   };
   // Execute our callback whenever window.postMessage is called
   window.addEventListener('message', onPostMessageReceived);
   return {
-    dispose: () => window.removeEventListener('message', onPostMessageReceived),
+    dispose: () => {
+      window.removeEventListener('message', onPostMessageReceived);
+      window.removeEventListener('message', checkForProtocolChange);
+    },
```

Comment:
we set up these dispose functions but never call them. We should remove the event listeners when we get the subscription termination pm, but I don't think we need to return a function here for any reason, right?